types/decimal.go

```diff
@@ -0,0 +1,324 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+// NOTE: never use new(Dec) or else we will panic unmarshalling into the
+// nil embedded big.Int
+type Dec struct {
+	*big.Int `json:"int"`
+}
+
+// number of decimal places
+const Precision = 10
```

Comment:
I assume these are base 10 decimal places. Have we benchmarked the performance gains we'd get with base 2 here? It may be worth considering. (Though I guess we could hard fork and change this #postlaunch)

Reply:
Check it out rat.Big doesn't store things in any bases - it only presents things in the desired base when you call getting a string... so I think it's already optimized! However I will quote that we determined during our call (Jae, you and myself I believe) that we cannot use base 2 due to some rounding problems for user interaction.  - but again I don't think this issue matters because optimized in `big.Int` -> https://golang.org/pkg/math/big/#Int.Text

Reply:
Your totally right! Thanks for the clarification